### DISCUSSION TITLE
Retrieving the available WLAN networks from the Android "Settings" App

### DISCUSSION THREAD

**Discussion by Howlinmoon** [appium/appium#20320] [Q&A]
Hi,
I am attempting to automate WLAN testing of a new Android device under development, and I'd like to use Appium.
My current requirement is to be able to retrieve the list of WLANs displayed under "Internet" in the "Settings" App.

I can navigate to "Internet" using Appium - but I can't figure out how to "scrape the screen" for lack of a better term.
When using Selenium to do something similar - I'd merely dump the source of the page and have my Python program parse out what I am looking for.

Is there a way to do something similar with Appium?

(I'd ask this is on the official Appium forum (discuss.appium.io) but the email verification routine seems to be busted so I can't signup)

Thanks for any suggestions, pointers etc

Jim

**Comment by mykola-mokhnach** [comment]
You can do the same in uia2 driver (e.g. dump the page source).

It could also be done faster by executing the following adb command, which lists available wifi networks: `adb shell cmd wifi list-scan-results`. You may use https://github.com/appium/appium-uiautomator2-driver?tab=readme-ov-file#mobile-shell to run it

**Comment by mykola-mokhnach** [comment]
There is more to that. `adb shell cmd wifi` shows the list of supported automation commands
